BubbleSort/BubbleSort.Tests/BubbleSortTests.cs

```diff
@@ -0,0 +1,169 @@
+namespace BubbleSort.Tests;
+
+public class Tests
+{
+    [TestCaseSource(typeof(TestData), nameof(TestData.TestCasesInt))]
+    public void TestSortInt(List<int> list, List<int> expectedResult)
+    {
+        var comparer = new ComparerInt();
+        var result = BubbleSort.Sort(list, comparer);
+        Assert.That(result, Is.EqualTo(expectedResult));
+    }
+
+    [TestCaseSource(typeof(TestData), nameof(TestData.TestCasesIntRevers))]
+    public void TestSortIntReversOrder(List<int> list, List<int> expectedResult)
+    {
+        var comparer = new ComparerIntRevers();
+        var result = BubbleSort.Sort(list, comparer);
+        Assert.That(result, Is.EqualTo(expectedResult));
+    }
+
+    [TestCaseSource(typeof(TestData), nameof(TestData.TestCasesString))]
+    public void TestSortString(List<string> list, List<string> expectedResult)
+    {
+        var comparer = new ComparerString();
+        var result = BubbleSort.Sort(list, comparer);
+        Assert.That(result, Is.EqualTo(expectedResult));
+    }
+
+    [TestCaseSource(typeof(TestData), nameof(TestData.TestCasesArray))]
+    public void TestSortString(List<int[]> list, List<int[]> expectedResult)
+    {
+        var comparer = new ComparerIntArray();
+        var result = BubbleSort.Sort(list, comparer);
+        Assert.That(result, Is.EqualTo(expectedResult));
+    }
+
+    [Test]
+    public void TestExceptions()
+    {
+        var comparer = new ComparerInt();
+        int[] list = [1, 2, 3];
+        Assert.Throws<ArgumentNullException>(() => BubbleSort.Sort(list, null));
+        Assert.Throws<ArgumentNullException>(() => BubbleSort.Sort(null, comparer));
+    }
+
+    public class ComparerInt : IComparer<int>
+    {
+        public int Compare(int x, int y)
+        {
+            if (x > y)
+            {
+                return 1;
+            }
+            else if (x == y)
+            {
+                return 0;
+            }
+            else
+            {
+                return -1;
+            }
+        }
+    }
+
+    public class ComparerIntRevers : IComparer<int>
+    {
+        public int Compare(int x, int y)
+        {
+            if (x > y)
+            {
+                return -1;
+            }
+            else if (x == y)
+            {
+                return 0;
+            }
+            else
+            {
+                return 1;
+            }
+        }
+    }
+
+    public class ComparerString : IComparer<string>
+    {
+        public int Compare(string x, string y)
+        {
+            return string.Compare(x, y);

```

Comment:
Кстати, справедливо, 
```suggestion
            return string.Compare(x, y, CultureInfo.InvariantCulture, CompareOptions.None);

```